// @unocss/preset-mini/dist/index.mjs
import { entriesToCss } from '@unocss/core';
import './shared/preset-mini.b41b4e0c.mjs';
export { p as parseColor } from './shared/preset-mini.81c09acb.mjs';
import { r as rules } from './shared/preset-mini.d4c3ecce.mjs';
export { c as colors } from './shared/preset-mini.e4b52abc.mjs';
import { t as theme } from './shared/preset-mini.7a539ada.mjs';
export { t as theme } from './shared/preset-mini.7a539ada.mjs';
import { v as variants } from './shared/preset-mini.aaf016bb.mjs';
import './shared/preset-mini.4af6c341.mjs';

const preflights = [
  {
    layer: "preflights",
    getCSS(ctx) {
      if (ctx.theme.preflightBase) {
        const css = entriesToCss(Object.entries(ctx.theme.preflightBase));
        return `*,::before,::after{${css}}::backdrop{${css}}`;
      }
    }
  }
];

const presetMini = (options = {}) => {
  options.dark = options.dark ?? "class";
  options.attributifyPseudo = options.attributifyPseudo ?? false;
  return {
    name: "@unocss/preset-mini",
    theme,
    rules,
    variants: variants(options),
    options,
    postprocess: options.variablePrefix && options.variablePrefix !== "un-" ? VarPrefixPostprocessor(options.variablePrefix) : void 0,
    preflights,
    prefix: options.prefix
  };
};
function VarPrefixPostprocessor(prefix) {
  return (obj) => {
    obj.entries.forEach((i) => {
      i[0] = i[0].replace(/^--un-/, `--${prefix}`);
      if (typeof i[1] === "string")
        i[1] = i[1].replace(/var\(--un-/g, `var(--${prefix}`);
    });
  };
}

export { presetMini as default, preflights, presetMini };
